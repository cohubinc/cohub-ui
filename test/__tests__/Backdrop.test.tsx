import React from "react";
import { mount } from "enzyme";

import { Backdrop } from "dist";

describe("Backdrop component", () => {
  it("renders", async () => {
    const wrapper = mount(<Backdrop open={true} focusTrapped={false} />);

    expect(wrapper).toMatchSnapshot();
  });
});
