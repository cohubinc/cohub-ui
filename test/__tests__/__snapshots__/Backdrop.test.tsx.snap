// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop component renders 1`] = `
<Backdrop
  containerClass=""
  focusTrapped={false}
  onClose={[Function]}
  open={true}
  showCloseIcon={false}
>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={
      Object {
        "closeButton": "closeButton",
        "modal": "modal",
        "overlay": "CohubBackdrop ",
      }
    }
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "modalCenter": "styles_modalCenter__L9F2w",
        "overlay": "styles_overlay__CLSq-",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <svg
        fill="none"
        height={44}
        viewBox="0 0 24 24"
        width={44}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 0.974332L11.025 0L6.00034 5.02532L0.975021 0L0 0.974332L5.02532 5.99966L0 11.025L0.975021 11.9993L6.00034 6.97399L11.025 11.9993L12 11.025L6.97468 5.99966L12 0.974332Z"
          fill="hsl(0, 0%, 100%)"
        />
      </svg>
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={false}
    onClose={[Function]}
    onEscKeyDown={[Function]}
    onOverlayClick={[Function]}
    open={true}
    role="dialog"
    showCloseIcon={false}
    styles={
      Object {
        "overlay": undefined,
      }
    }
  >
    <Portal
      containerInfo={
        <div>
          <div
            class="styles_overlay__CLSq- CohubBackdrop styles_transitionEnter__3j_-a styles_transitionEnterActive___eQs7"
          >
            <div
              aria-modal="true"
              class="styles_modal__gNwvD modal"
              role="dialog"
            />
          </div>
        </div>
      }
    >
      <CSSTransition
        appear={true}
        classNames={
          Object {
            "appear": "styles_transitionEnter__3j_-a",
            "appearActive": "styles_transitionEnterActive___eQs7",
            "enter": "styles_transitionEnter__3j_-a",
            "enterActive": "styles_transitionEnterActive___eQs7",
            "exit": "styles_transitionExit__1KmEf",
            "exitActive": "styles_transitionExitActive__1nQXw",
          }
        }
        in={true}
        onEntered={[Function]}
        onExited={[Function]}
        timeout={500}
      >
        <Transition
          appear={true}
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={500}
          unmountOnExit={false}
        >
          <div
            className="styles_overlay__CLSq- CohubBackdrop "
            onClick={[Function]}
          >
            <div
              aria-modal="true"
              className="styles_modal__gNwvD modal"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="dialog"
            />
          </div>
        </Transition>
      </CSSTransition>
    </Portal>
  </Modal>
</Backdrop>
`;
